samples/valueentity-counter/src/test/java/com/example/domain/CounterTest.java

```diff
@@ -0,0 +1,102 @@
+        entity.increaseWithReply(command, context);
+
+        Mockito.verify(context).updateState(CounterDomain.CounterState.newBuilder().setValue(42).build());
+    }
+
+    @Test
+    public void increaseWithPriorState() {
+        entity = new CounterImpl(entityId);
+        CommandContext<CounterDomain.CounterState> context = getCounterStateCommandContext(13);
+
+        CounterApi.IncreaseValue command = CounterApi.IncreaseValue.newBuilder().setValue(42).build();
+        entity.increaseWithReply(command, context);
+
+        Mockito.verify(context).updateState(CounterDomain.CounterState.newBuilder().setValue(13 + 42).build());
+    }
+
+    @Test
+    public void increaseShouldFailWithNegativeValue() {
+        entity = new CounterImpl(entityId);
+        CommandContext<CounterDomain.CounterState> context = getCounterStateCommandContext(27);
+        Mockito.when(context.fail(anyString()))
+                .thenReturn(new MockedContextFailure());
+
+        assertThrows(MockedContextFailure.class, () -> {
+            CounterApi.IncreaseValue command = CounterApi.IncreaseValue.newBuilder().setValue(-2).build();
+            entity.increaseWithReply(command, context);
+        });
+    }
+
+    @Test
+    public void decreaseNoPriorState() {
+        entity = new CounterImpl(entityId);
+        CommandContext<CounterDomain.CounterState> context = contextWithoutState();
+
+        CounterApi.DecreaseValue command = CounterApi.DecreaseValue.newBuilder().setValue(42).build();
+        entity.decreaseWithReply(command, context);
+
+        Mockito.verify(context).updateState(CounterDomain.CounterState.newBuilder().setValue(-42).build());
+    }
+    
+    @Test
+    public void resetTest() {
+        entity = new CounterImpl(entityId);
+        CommandContext<CounterDomain.CounterState> context = getCounterStateCommandContext(13);
+
+        CounterApi.ResetValue command = CounterApi.ResetValue.newBuilder().build();
+        entity.reset(command, context);
+
+        Mockito.verify(context).updateState(CounterDomain.CounterState.newBuilder().setValue(0).build());
+    }
+    
+    @Test
+    public void getCurrentCounterTest() {
+        entity = new CounterImpl(entityId);
+
+        CommandContext<CounterDomain.CounterState> context = getCounterStateCommandContext(13);
+
+        CounterApi.GetCounter command = CounterApi.GetCounter.newBuilder().build();
+        CounterApi.CurrentCounter reply = entity.getCurrentCounter(command, context);
+
+        assertThat(reply.getValue(), is(13));
+    }
+
+    private CommandContext<CounterDomain.CounterState> contextWithoutState() {
+        CommandContext<CounterDomain.CounterState> context = Mockito.mock(CommandContext.class);
+        Mockito.when(context.getState()).thenReturn(Optional.empty());
+        return context;
+    }
+
+    private CommandContext<CounterDomain.CounterState> getCounterStateCommandContext(int value) {
```

Comment:
`getCounterStateCommandContext` sounds like we are getting the `CommandContext` from the `CounterState` or the `Context` from the `CounterStateCommand`. 

What about `contextWithState(int counterValue)`?
It aligns better with `contextWithoutState`.